demo/api/src/importer/pipes/parsers/csv-data-transformer.pipe.ts

```diff
@@ -0,0 +1,208 @@
+import { LoggerService } from "@nestjs/common";
+import { CsvColumnMetadata } from "@src/importer/decorators/csv-column.decorator";
+import { FieldTransformerData } from "@src/importer/decorators/field-transformer.decorator";
+import { ImporterEntityClass } from "@src/importer/entities/base-target.entity";
+import {
+    columnIsBoolean,
+    columnIsDate,
+    columnIsFloat,
+    columnIsInteger,
+    isNullable,
+    TargetEntityProperties,
+    TargetEntityProperty,
+} from "@src/importer/entities/target-entity.utils";
+import { plainToInstance } from "class-transformer";
+import { validate } from "class-validator";
+import { parse, parseISO } from "date-fns";
+import { Transform, TransformCallback } from "stream";
+
+import { ImporterPipe, PipeData, ValidationError } from "../importer-pipe.type";
+
+type ValidateAndTransformValueProperties = {
+    value: string;
+    propertyMetadata: TargetEntityProperty;
+    field: CsvColumnMetadata;
+};
+
+type DataTransformOptions = {
+    fields: CsvColumnMetadata[];
+    targetEntityProperties: TargetEntityProperties;
+    entityDateFormatString?: string;
+    fieldTransformers?: FieldTransformerData[];
+};
+
+type PossibleValueTypes = string | number | boolean | Date | undefined | null;
+
+export class CsvDataTransformerPipe implements ImporterPipe {
+    constructor(private readonly options: DataTransformOptions, private readonly entity: ImporterEntityClass) {
+        this.options = options;
+    }
+
+    getPipe(logger: LoggerService) {
+        return new CsvDataTransformer(this.options, logger, this.entity);
+    }
+}
+
+type ParserPipeData = Record<string, string>;
+
+export class CsvDataTransformer extends Transform {
```

Comment:
This class does not work with CsvData but with a simple js-object. Could this also work for any other source-format, json for example? You could consider renaming the `@CsvColumn()`-decorator to `TransformData()`, or to something other more generic...